src/components/auth/SignUpScreen.js

```diff
@@ -1,290 +1,316 @@
+                  />
+                </Item>
+                <Item style={styles.itemStyle}>
+                  <Input
+                    placeholder={i18n.t("Password")}
+                    placeholderTextColor={COLORS.forestgreen}
+                    returnKeyType="next"
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    secureTextEntry={this.state.hidePassword}
+                    ref="ThirdInput"
+                    onSubmitEditing={(event) => {
+                      this.refs.FourthInput._root.focus();
+                    }}
+                    onChangeText={(value) =>
+                      this.onChangeText("password", value)
+                    }
+                  />
+
+                  <TouchableOpacity onPress={() => this.handleHidePassword()}>
+                    <Image
+                      source={
+                        this.state.hidePassword == false
+                          ? show_hide_icons[0]
+                          : show_hide_icons[1]
+                      }
+                      style={styles.showHideIcon}
                     />
-                    <Ionicons style={styles.iconStyle2} name="ios-eye" onPress={() => this.handleHidePassword2()}/>
-                  </Item>
-                  {/* SignUp Button */}
-                  <TouchableOpacity
-                    onPress={() => this.handleSignUp()}
-                    disabled={this.state.isLoading}
-                    style={styles.buttonStyle1}>
-                    <Text style={styles.buttonText1}>
-                      {i18n.t('Sign Up')}
-                    </Text>
                   </TouchableOpacity>
+                  {/* <Ionicons
+                    style={
+                      this.state.hidePassword == false
+                        ? styles.showIcon
+                        : styles.hideIcon
+                    }
+                    name="ios-eye"
+                    onPress={() => this.handleHidePassword()}
+                  /> */}
+                </Item>
+                <Item style={styles.itemStyle}>
+                  <Input
+                    placeholder={i18n.t("Confirm Password")}
+                    placeholderTextColor={COLORS.forestgreen}
+                    returnKeyType="go"
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    secureTextEntry={this.state.hidePassword}
+                    ref="FourthInput"
+                    onChangeText={(value) =>
+                      this.onChangeText("password_confirmation", value)
+                    }
+                  />
+                </Item>
+
+                {/* SignUp Button */}
+                <TouchableOpacity
+                  onPress={() => this.handleSignUp()}
+                  disabled={this.state.isLoading}
+                  style={styles.submitButton}>
+                  <Text style={styles.submitLabel}>{i18n.t("Sign Up")}</Text>
```

Comment:
Add a {!this.state.isLoading &&... basically, the activityindicator replaces the text.

Reply:
does this work?

Reply:
Is isLoading, show only the activityIndicator, if !isLoading show only the text

Reply:
{!this.state.isLoading && (
                    <Text style={styles.submitLabel}>{i18n.t("Sign Up")}</Text>
                  )}

Reply:
Exactly